tests/test_content_manifest.py

```diff
@@ -152,29 +152,63 @@ def test_process_pip(default_request, default_toplevel_purl):
     assert cm._pip_data == expected_contents
 
 
+def test_process_gitsubmodule(default_request):
+    submodule_commit_id = "58c88e4952e95935c0dd72d4a24b0c44f2249f5b"
+    pkg = Package.from_json(
+        {
+            "name": "tour",
+            "type": "git-submodule",
+            "version": f"https://github.com/testrepo/tour.git#{submodule_commit_id}",
+        }
+    )
+    pkg.id = 1
+    expected_purl = "pkg:github/testrepo/tour@58c88e4952e95935c0dd72d4a24b0c44f2249f5b"
+
+    cm = ContentManifest(default_request)
+
+    # emulate to_json behavior to setup internal packages cache
+    cm._gitsubmodule_data.setdefault(
+        pkg.id, {"purl": expected_purl, "dependencies": [], "sources": []}
+    )
+
+    expected_contents = {pkg.id: {"purl": expected_purl, "dependencies": [], "sources": []}}
+
+    assert cm._gitsubmodule_data
+    assert pkg.id in cm._gitsubmodule_data
+    assert cm._gitsubmodule_data == expected_contents
+
+
 @pytest.mark.parametrize(
-    "package", [None, {"name": "example.com/org/project", "type": "go-package", "version": "1.1.1"}]
+    "packages",
+    [
+        [
+            None,
+            {"name": "example.com/org/project", "type": "go-package", "version": "1.1.1"},
+            {
+                "name": "tour",
+                "type": "git-submodule",
+                "version": (
+                    "https://github.com/testrepo/tour.git#58c88e4952e95935c0dd72d4a24b0c44f2249f5b"
+                ),
+            },
+        ]
+    ],
 )
-def test_to_json(app, package):
+@mock.patch("cachito.web.content_manifest.ContentManifest.generate_icm")
+def test_to_json(mock_generate_icm, app, packages):
     request = Request()
     cm = ContentManifest(request)
 
     image_contents = []
-    if package:
-        pkg = Package.from_json(package)
-        request.packages.append(pkg)
-        content = {"purl": pkg.to_purl(), "dependencies": [], "sources": []}
-        image_contents.append(content)
-
-    expected = {
-        "metadata": {
-            "icm_version": 1,
-            "icm_spec": ContentManifest.json_schema_url,
-            "image_layer_index": -1,
-        },
-        "image_contents": image_contents,
-    }
-    assert cm.to_json() == expected
+    for package in packages:
+        if package:
+            pkg = Package.from_json(package)
+            request.packages.append(pkg)
+            content = {"purl": pkg.to_purl(), "dependencies": [], "sources": []}
+            image_contents.append(content)
+    res = cm.to_json()
+    mock_generate_icm.assert_called_once_with(image_contents)
+    assert res == mock_generate_icm.return_value
```

Comment:
Why were the changes in this test needed?

Reply:
Since, I passed an additional package there, I'd need to update the `expected` to have deep sort run on it to make it an Ordered dict probably. And we already have a separate test beneath `test_generate_icm`. And this test is for `to_json` so why not just mock `generate_icm`?

Reply:
But the `packages` param was originally just `package`, right? It is a good idea to also test with multiple packages, but then I would like to keep the original test cases as well (empty list, list with just 1 package).

+1 to mocking `generate_icm`